lib/index.js

```diff
@@ -49,10 +49,28 @@ class SpikeCollections {
         cwd: compiler.options.context,
         nodir: true,
         realpath: true,
-        ignore: compiler.options.spike.ignore
+        ignore: this.util.getSpikeOptions().ignore
       })
     })
 
+    // all files as a straight array, no categories
+    this.allFiles = reduce(this.files, (m, v) => m.push(...v) && m, [])
+
+    // add loader alias so that frontmatter loader can be resolved
+    const resolveLoader = compiler.options.resolveLoader
+    if (!resolveLoader.alias) resolveLoader.alias = {}
+    resolveLoader.alias.frontmatter = path.join(__dirname, 'frontmatter_loader.js')
+
+    // add frontmatter loader with pattern converted to regex
+    compiler.options.module.rules.push({
+      test: this.util.pathsToRegex(this.allFiles),
+      use: [{
+        loader: 'frontmatter',
+        // NOTE: this *could* be configurable
+        options: { _spikeExtension: 'html' }
+      }]
+    })
```

Comment:
So this was the biggest change here. Webpack2 changed the way that modules resolve their loaders, which ended up changing the order in which this happens in their compile process. Previously, you could add a new loader (now rule) in the `make` plugin hook and this would be before webpack resolved the module's path against its loaders' tests and figured out which loaders to bind to it. With webpack2, this is no longer the case -- the loaders are bound earlier in the process. Shifting this extra loader push up into the initial apply method means it runs as early as possible and ensures that it runs before loaders and bound to modules.